package.json

```
@@ -108,7 +108,8 @@
     "fmt": "prettier --write .*.json *.{json,md,js} **/*.ts './{.circleci,.github,build,doc,lib,locales,test,typings}/**/*.{json,md,js,ts,html}'",
     "prepare": "husky install",
     "prebuild": "node ./build/check-node-version.js",
-    "pretest": "node ./build/check-node-version.js"
+    "pretest": "node ./build/check-node-version.js",
+    "postbuild": "prettier --write ./locales/_template.json ./doc/rule-descriptions.md"
```

I see no reason why we shouldn't format everything here.

```suggestion
    "postbuild": "prettier --write ."
```

---

Doing this will add so much noise to prs. We only run prettier on files changed so we only include files that were changed.

---

Doing this will cause 1 PR to generate noise; after that, everything will be formatted (therefore no more noise).

---

This also removes the need to run prettier on precommit. I'd rather we get this small change in to fix a problem and have a larger discussion about how we want to do prettier in a separate ticket.

---

This does not remove the need for the precommit hook. It's very possible to change a file in this repository without running `npm run build`.

What is your hesitation here? I don't understand why this change isn't being accepted.

---

A few reasons:

1. running prettier on every file as part of this pr is going to cause merge conflicts to every open pr, especially the pr which refactors the entire codebase away from our visibility functions. I've been waiting for prs to be settled before running prettier on the entire code base. 
2. this pr, which was a 5 minute change, has now taken multiple days in order to fix something that should have been trivial. I'd like to fix the problem before scope creeping to a larger discussion about how prettier is run on the repo